### PR TITLE
Add some tests covering main logic

### PR DESCRIPTION
**Pull request by fyrchik**
See #1 .
Also fix 2 bugs discovered during tests.
Yet to be done:
- change view tests
- some corner cases with invalid fields like `NextConsensus`

### CODE REVIEW

File: dbft_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	currHeight uint32
+	currHash   util.Uint256
+}
+
+func TestDBFT_OnStartPrimarySendPrepareRequest(t *testing.T) {
+	const validatorsCount = 7
+
+	s := newTestState(2, validatorsCount)
+
+	t.Run("backup sends nothing on start", func(t *testing.T) {
```
Comment (volekerb) [inline dbft_test.go]: really like when tests look like this. We need to use the same approach in neo-go

File: dbft_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
+package dbft
+
+import (
+	"crypto/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
```
Comment (volekerb) [inline dbft_test.go]: nit:dunno if this empty space required

File: dbft_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		count:   count,
+		ch:      make(chan Payload, 1),
+	}
+
+	s.privs, s.pubs = getTestValidators(count)
+
+	return s
+}
+
+func (s *testState) getOptions() []Option {
```
Comment (volekerb) [inline dbft_test.go]: nit: in each func you use "Test" but here not. Probably it's better to stick to one way of writing test function

File: dbft.go [hunk 4]
```diff
@@ -125,12 +125,12 @@ func (d *DBFT) InitializeConsensus(view byte) {
 }
 
 func (d *DBFT) OnTransaction(tx block.Transaction) {
-	// d.Logger.Debug("OnTransaction",
```
Comment (volekerb) [inline dbft.go]: this change looks like not relevant to commit message

File: dbft_test.go (hunk 3 above)
Comment (fyrchik) [reply dbft_test.go]: Because it is a method of a test struct.

File: dbft_test.go [hunk 5]
```diff
@@ -54,6 +65,94 @@ func TestDBFT_OnStartPrimarySendPrepareRequest(t *testing.T) {
 	})
 }
 
+func TestDBFT_OnReceiveRequestSendResponse(t *testing.T) {
+	const (
+		validatorCount = 7
```
Comment (volekerb) [inline dbft_test.go]: why do you use 7 here?

File: dbft_test.go [hunk 6]
```diff
@@ -228,7 +281,7 @@ func newTestState(myIndex int, count int) *testState {
 	s := &testState{
 		myIndex: myIndex,
 		count:   count,
-		ch:      make(chan Payload, 1),
+		ch:      make(chan Payload, 4),
```
Comment (volekerb) [inline dbft_test.go]: hm it's changed for all tests to 4. And interesting why is that so =)

**Comment by roman-khimov** [comment]
I'd split f7bb64808625631b429ad45da653b09cf167046e into two separate commits for the respective issues.

File: dbft_test.go (hunk 6 above)
Comment (fyrchik) [reply dbft_test.go]: I initially approached this the wrong way. Testing routine should never hang on channel send so I have replaced this channel with slice and removed magic constants.

File: dbft_test.go (hunk 5 above)
Comment (fyrchik) [reply dbft_test.go]: It plays no role here.
